applications/luci-app-olsr-viz/htdocs/luci-static/resources/view/olsr-viz/olsr-viz-view.js

```diff
@@ -0,0 +1,173 @@
+'use strict';
+'require uci';
+'require view';
+'require poll';
+'require ui';
+'require rpc';
+
+
+return view.extend({
+	callGetOlsrVizData: rpc.declare({
+		object: 'olsrvizinfo',
+		method: 'getolsrvizdata'
+	}),
+
+	fetch_jsoninfo: function () {
+		var jsonreq4 = '';
+		var json;
+		var data;
+		var self = this;
+		return new Promise(function (resolve, reject) {
+			L.resolveDefault(self.callGetOlsrVizData(), {})
+				.then(function (res) {
+					json = res;
+					data = json.jsonreq4;
+					resolve([data]);
+				})
+				.catch(function (err) {
+					console.error(err);
+					reject([null]);
+				});
+		});
+	},
+
+	action_olsr_viz: function () {
+		var self = this;
+		return new Promise(function (resolve, reject) {
+			self
+				.fetch_jsoninfo()
+				.then(function ([data]) {
+					var result = { viz_data: data };
+					resolve(result);
+				})
+				.catch(function (err) {
+					reject(err);
+				});
+		});
+	},
+
+	load: function () {
+		var self = this;
+		poll.add(function () {
+			self.render();
+		}, 15);
+		return new Promise(function (resolve, reject) {
+			var script = E('script', { 'type': 'text/javascript' });
+			script.onload = function () {
+				resolve();
+			};
+			script.onerror = function (err) {
+				reject(err);
```

Comment:
You can shorten this to just `script.onerror = reject;` - we don't care about the actual return values or arguments, just that the resolve function is invoked.